assets/ca/ca.querytranslator.js

```diff
@@ -286,56 +310,94 @@ var caUI = caUI || {};
+	 * Use the given `queryValue` to assign a `value` and `condition` to the given `rule`.
+	 * @param {Object} rule
+	 * @param {String} min
+	 * @param {String} max
+	 * @param {Boolean} negation
+	 */
+	assignOperatorAndRange = function (rule, min, max, negation) {
+        rule.value = [ min, max ];
+        rule.operator = negation ? 'not_between' : 'between';
+	};
+
+	/**
 	 * Parse the given array of tokens into a tree structure for the query builder.
 	 * @param {Array} tokens
 	 * @return {Object}
 	 */
 	tokensToRuleSet = function (tokens) {
-		var rule, condition, negation, wildcardPrefix, wildcardSuffix, word, ruleSet;
+		var rule, condition, negation, wildcardPrefix, wildcardSuffix, min, max, word, ruleSet;
 		ruleSet = {
 			condition: undefined,
 			rules: []
 		};
 		skipWhitespace(tokens);
-		// End this recursion when the string is finished, or when we reach a right parenthesis.
-		while (tokens.length > 0 && tokens[0].type !== TOKEN_RPAREN) {
-			if (isNextToken(tokens, TOKEN_LPAREN)) {
-				// Explicitly nested rule set: recursion.
-				rule = tokensToRuleSet(tokens);
-				assertNextToken(tokens, TOKEN_RPAREN);
-			} else if (tokens[0].type !== TOKEN_RPAREN) {
-				// Standard rule, with a field, operator and value.
-				rule = {};
-				rule.field = rule.id = assertNextToken(tokens, TOKEN_WORD).value;
-				assertNextToken(tokens, TOKEN_COLON);
-				negation = isNextToken(tokens, TOKEN_NEGATION);
-				wildcardPrefix = isNextToken(tokens, TOKEN_WILDCARD);
-				word = assertNextToken(tokens, TOKEN_WORD);
-				wildcardSuffix = isNextToken(tokens, TOKEN_WILDCARD);
-				assignOperatorAndValue(rule, word.value, negation, wildcardPrefix, wildcardSuffix);
-			}
-			skipWhitespace(tokens);
-			if (rule) {
-				ruleSet.rules.push(rule);
-				if (tokens.length > 0 && tokens[0].type !== TOKEN_RPAREN) {
-					// Process the next condition ("AND" / "OR").
-					condition = assertCondition(tokens).value;
-					// Assign the first condition to the rule set.
-					ruleSet.condition = ruleSet.condition || condition;
-					if (condition !== ruleSet.condition) {
-						// We have something like "A AND B OR C" in the query.  This is interpreted as "(A AND B) OR C".
-						// The "AND" and "OR" conditions are given equal precedence, so the parentheses are always
-						// around the left-most set of filters with matching condition.  This is implemented by pushing
-						// the existing rule set down a level in the hierarchy, and continuing processing from the new
-						// parent.
-						ruleSet = {
-							condition: condition,
-							rules: [ ruleSet ]
-						};
+		if (isRawSearchText(tokens)) {
+			// Special case: a sequence of only words (and whitespace) should be treated as a single, full text search.
+			ruleSet.condition = 'AND';
+			ruleSet.rules.push({
+				id: FIELD_FULLTEXT,
+				field: FIELD_FULLTEXT,
+				operator: 'equal',
+				value: $.map(tokens, function (token) {
+					return token.type === TOKEN_WHITESPACE ? ' ' : token.value;
```

Comment:
As a side-effect this collapses whitespace, I think that's acceptable, and probably actually preferred, since whitespace doesn't affect the search.